### PR TITLE
optimize service extraddrs

### PR DESCRIPTION
**Pull request by zhengbingxian**
**Please provide a description of this PR:**

Kubernetes service field "clusterIPs"  hold  a maximum of two entries (dual-stack IPs).  
And ensure clusterIPs[0] and clusterIP have the same value.

So extraddrs can optimize.
according to the [clusterips definition](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/)

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/conversion.go [hunk 1]
```diff
@@ -71,13 +71,8 @@ func ConvertService(svc corev1.Service, domainSuffix string, clusterID cluster.I
...
 		addr = svc.Spec.ClusterIP
-		if len(svc.Spec.ClusterIPs) > 0 {
-			for _, ip := range svc.Spec.ClusterIPs {
-				// exclude the svc.Spec.ClusterIP
-				if ip != addr {
-					extrAddrs = append(extrAddrs, ip)
-				}
-			}
+		if len(svc.Spec.ClusterIPs) > 1 {
+			extrAddr = svc.Spec.ClusterIPs[1]
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/conversion.go:75]: It is not guranteed the order of the ip addresses. 

```
This field may hold a maximum of two entries (dual-stack IPs, in either order). These IPs must correspond to the values of the ipFamilies field. Both clusterIPs and ipFamilies are governed by the ipFamilyPolicy field. More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
```

**Review by hzxuzhonghu — COMMENTED** [review]
/ok-to-test

File: pilot/pkg/serviceregistry/kube/conversion.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/conversion.go:75]: You are right
```
If this field is specified, clients must ensure that clusterIPs[0] and clusterIP have the same value.
```

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by zhengbingxian** [comment]
@hzxuzhonghu   
Could you please add release-notes-none labels and ok-to-test labels? Or is there anything  left should I  to do?

**Comment by hzxuzhonghu** [comment]
please fix the ut

**Pull request merged**